Src/AutoFixture/ElementsBuilder.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Ploeh.AutoFixture.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Draws a random element from the given collection
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public sealed class ElementsBuilder<T> : ISpecimenBuilder
+    {
+        private readonly ICollection<T> elements;
+        private readonly RandomNumericSequenceGenerator sequence;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ElementsBuilder{T}"/> class.
+        /// </summary>
+        /// <param name="elements">The collection from which elements should be drawn from.
+        /// It must contain at least two element.</param>
+        public ElementsBuilder(ICollection<T> elements)
+        {
+            if (elements == null)
+                throw new ArgumentNullException("elements");
+
+            if (elements.Count < 2)
```

Comment:
Why is a collection with a single element not allowed?

Granted, a singleton collection would be degenerate, and result in deterministic behaviour, but it'd still work.

From a user experience perspective, I think it's quite clear than when you have a function that select a random element from a list, and that list only has one element, then the return value will always be that single element.

FWIW, [FsCheck's](http://fscheck.github.io/FsCheck) `Gen.elements` allows the use of a singleton list, as demonstrated by this passing property:

```F#
[<Property>]
let test (i : int) = Prop.forAll (Gen.elements [i] |> Arb.fromGen) ((=) i)
```